x/shared/helpers/service_test.go

```diff
@@ -1,78 +1,181 @@
 package helpers
 
-import "testing"
+import (
+	"testing"
+
+	sharedtypes "github.com/pokt-network/poktroll/x/shared/types"
+)
+
+func TestIsValidService(t *testing.T) {
+	tests := []struct {
+		testCase string
+		id       string
+		name     string
+		expected bool
+	}{
+		{
+			testCase: "Valid ID and Name",
+			id:       "Service1",
+			name:     "Valid Service Name",
+			expected: true,
+		},
+		{
+			testCase: "Valid ID and empty Name",
+			id:       "Srv",
+			name:     "", // Valid because the service name can be empty
+			expected: true,
+		},
+		{
+			testCase: "ID exceeds max length",
+			id:       "TooLongId123", // Exceeds maxServiceIdLength
+			name:     "Valid Name",
+			expected: false,
+		},
+		{
+			testCase: "Name exceeds max length",
+			id:       "ValidID",
+			name:     "This service name is way too long to be considered valid since it exceeds the max length",
+			expected: false,
+		},
+		{
+			testCase: "Empty ID",
+			id:       "", // Invalid because the service ID cannot be empty
+			name:     "Valid Name",
+			expected: false,
+		},
+		{
+			testCase: "Invalid characters in ID",
+			id:       "ID@Invalid", // Invalid character '@'
+			name:     "Valid Name",
+			expected: false,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.testCase, func(t *testing.T) {
+			service := &sharedtypes.Service{
+				Id:   test.id,
+				Name: test.name,
+			}
+			result := IsValidService(service)
+			if result != test.expected {
+				t.Errorf("Test Case '%s' - IsValidService() with Id: '%s', Name: '%s', expected %v, got %v",
+					test.testCase, test.id, test.name, test.expected, result)
+			}
+		})
+	}
+}
 
 func TestIsValidServiceId(t *testing.T) {
 	tests := []struct {
+		testCase string
```

Comment:
NIT: wdyt about `name` instead?

```suggestion
		name string
```

Reply:
I'm noticing that ignite generated tests seem to use the word `desc`:

![image](https://github.com/pokt-network/poktroll/assets/600733/ed0b0a61-3a86-4020-96eb-d7953a46bcd0)